text/0281-es5-getters.md

```diff
@@ -0,0 +1,180 @@
+- Start Date: 2017-12-12
+- RFC PR: https://github.com/emberjs/rfcs/pull/281
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Install ES5 getters for computed properties on object prototypes, thus
+eliminating the need to use `this.get()` or `Ember.get()` to access them.
+
+Before:
+
+```js
+import Object, { computed } from '@ember/object';
+
+const Person = Object.extend({
+  fullName: computed('firstName', 'lastName', function() {
+    return `${this.get('firstName')} ${this.get('lastName')}`;
+  })
+});
+
+let chancancode = Person.create({ firstName: 'Godfrey', lastName: 'Chan' });
+
+chancancode.get('fullName'); // => 'Godfrey Chan'
+
+chancancode.set('firstName', 'ʎǝɹɟpo⅁');
+
+chancancode.get('fullName'); // => 'ʎǝɹɟpo⅁ Chan'
+
+let { firstName, lastName, fullName } = chancancode.getProperties('firstName', 'lastName', 'fullName');
+```
+
+After:
+
+```js
+import Object, { computed } from "@ember/object";
+
+const Person = Object.extend({
+  fullName: computed('firstName', 'lastName', function() {
+    return `${this.firstName} ${this.lastName}`;
+  })
+});
+
+let chancancode = Person.create({ firstName: 'Godfrey', lastName: 'Chan' });
+
+chancancode.fullName; // => 'Godfrey Chan'
+
+chancancode.set('firstName', 'ʎǝɹɟpo⅁');
+
+chancancode.fullName; // => 'ʎǝɹɟpo⅁ Chan'
+
+let { firstName, lastName, fullName } = chancancode; // No problem!
+```
+
+# Motivation
+
+Ember inherited its computed properties functionality from [SproutCore](http://guides.sproutcore.com/core_concepts_kvo.html).
+The feature was designed at a time before [ES5 getters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get)
+were widely available. This necessitated using a special function such as
+`this.get()` or `Ember.get()` to access the values of computed properties.
+
+Since all of our [target browsers](https://github.com/emberjs/rfcs/pull/252)
+support ES5 getters now, we can drop the need of this special function,
+improving developer egornomics and interoperability between other libraries
+and tooling (such as TypeScript).
+
+Note that at present, using `this.set()` or `Ember.set()` is still mandatory
+for the property to recompute properly. In the future, we might be able to
+loosen this requirement, perhaps with the help of ES5 setters. However, that
+would require more design and is out-of-scope for this RFC.
+
+`this.get()` and `Ember.get()` will still work. This RFC does not propose
+removing or deprecating them in the near term. They support other use cases
+that ES5 getters does not, such as "safe" path chaining (`get('foo.bar.baz')`)
```

Comment:
For the curious: Babel 7 will support [optional chaining](https://github.com/babel/babel/tree/master/packages/babel-plugin-proposal-optional-chaining).